rejuvenate.lua

```diff
@@ -24,6 +24,36 @@ function rejuvenate(unit, force, dry_run, age)
         unit.old_year = new_birth_year + 160
     end
     if unit.profession == df.profession.BABY or unit.profession == df.profession.CHILD then
+        if unit.profession == df.profession.BABY then
+            local leftoverUnits = {}
+            local shiftedLeftoverUnits = {}
+            -- create a copy
+            local babyUnits = df.global.world.units.other.ANY_BABY
+            -- create a new table with the units that aren't being removed in this iteration
+            for _, v in ipairs(babyUnits) do
+                if not v.id == unit.id then
+                    table.insert(leftoverUnits, v)
+                end
+            end
+            -- create a shifted table of the leftover units to make up for lua tables starting with index 1 and the game starting with index 0
```

Comment:
please use the `erase` method on the `world.units.all` vector for this. this method directly calls the appropriate C++ method on the underlying vector instead of rewriting the vector

Reply:
you can also avoid finding the index entirely by using `utils.erase_sorted(df.global.world.units.other.ANY_BABY, unit, 'id')`

Reply:
> you can also avoid finding the index entirely by using `utils.erase_sorted(df.global.world.units.other.ANY_BABY, unit, 'id')`

do note that classification subvectors (such as `ANY_BABY`) are not guaranteed to be sorted. the `categorize` methods simply insert the unit/item/building at the end of the classified subvector. using tools that expect these vectors to be sorted may result in unexpected behavior

Reply:
Thanks for the heads up. I had trouble with lua doing a 1 based index vs C++ (as well as what I'm used to from other languages) doing a 0 based index and then that translating back into the game vector. I'll update with these fixes later this week/this weekend. Appreciate the suggestions